### PR TITLE
ref #187 Include requestAuthorization in Skygear Push Basics

### PR DESCRIPTION
**Pull request by david90**
Added a guide on iOS specific push set up about `requestAuthorization` for displaying push notifications.

### CODE REVIEW

File: push-notifications/basics/ios.md [hunk 1]
```diff
@@ -58,9 +69,65 @@ func application(_ application: UIApplication, didFinishLaunchingWithOptions lau
...
+```obj-c
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+
+
+	UNUserNotificationCenter* center = [UNUserNotificationCenter currentNotificationCenter];
+	[center requestAuthorizationWithOptions:(UNAuthorizationOptionAlert + UNAuthorizationOptionSound)
+	   completionHandler:^(BOOL granted, NSError * _Nullable error) {
+	      // Enable or disable features based on authorization.
+	}];
```
Comment (carmenlau) [inline push-notifications/basics/ios.md]: Please fix the indention.

File: push-notifications/basics/ios.md [hunk 2]
```diff
@@ -58,9 +69,65 @@ func application(_ application: UIApplication, didFinishLaunchingWithOptions lau
...
+    // This will prompt the user for permission to send remote notification
+    [application registerForRemoteNotifications];
+
+    // Other application initialization logic here
+
+    return YES;
+}
+
+
+
```
Comment (carmenlau) [inline push-notifications/basics/ios.md]: I found that the code is quite duplicate from the above section, excepts adding `requestAuthorizationWithOptions` call. Do you think we just update the first block of sample code?

File: push-notifications/basics/ios.md [hunk 3]
```diff
@@ -58,9 +69,65 @@ func application(_ application: UIApplication, didFinishLaunchingWithOptions lau
...
+```
+
+```swift
+let center = UNUserNotificationCenter.current()
+center.requestAuthorization(options: [.alert, .sound]) { (granted, error) in
+    // Enable or disable features based on authorization
+}
+
```
Comment (carmenlau) [inline push-notifications/basics/ios.md]: It would be better the swift version should be the same as obj-c version.

**Review by carmenlau — CHANGES_REQUESTED** [review]
Thanks a lot!! @david90

**Comment by david90** [comment]
@carmenlau  Updated!

File: push-notifications/basics/ios.md [hunk 4]
```diff
@@ -49,18 +83,27 @@ func application(_ application: UIApplication, didFinishLaunchingWithOptions lau
         // Anything you want to do in the callback can be added here
     }
 
-    // This will prompt the user for permission to send remote notification
+    let center = UNUserNotificationCenter.current()
+    center.requestAuthorization(options: [.alert, .sound]) { (granted, error) in
+    // This will prompt the user for permission to send remote notification    
     application.registerForRemoteNotifications()
+    
+    // Enable or disable features based on authorization
```
Comment (carmenlau) [inline push-notifications/basics/ios.md]: Missing closing `}` for `requestAuthorization`, and please update the identical. Other looks fine for me, thanks @david90

**Review by carmenlau — CHANGES_REQUESTED** [review]

**Comment by david90** [comment]
updated

**Comment by carmenlau** [comment]
Thanks @david90!

**Pull request merged**